client/main/vareditor/datavarwidget.js

```diff
@@ -128,19 +132,107 @@ const DataVarWidget = Backbone.View.extend({
             this._enableDisableMoveButtons();
         }
     },
-    _setLevels(levels) {
+    _setLevels(measureType, levels) {
         if ( ! this.attached)
             return;
-        this.$levelItems.off('click');
-        this.$levels.empty();
+
+        if (levels.length === 0)
+            this.$levels.empty();
+        else
+            this.$levelItems.remove(":gt(" + (levels.length - 1) + ")");
```

Comment:
in regards to detaching the remove method detaches the event listeners for you. In regards to the other stuff, I don't understand what you mean, you saying i should use a for loop and remove them individually?